cmd/kube-dns/app/server.go

```diff
@@ -163,55 +160,12 @@ func setupSignalHandlers() {
 	}()
 }
 
-func validateHostAndPort(hostAndPort string) error {
-	host, port, err := net.SplitHostPort(hostAndPort)
-	if err != nil {
-		return err
-	}
-	if ip := net.ParseIP(host); ip == nil {
-		return fmt.Errorf("bad IP address: %s", host)
-	}
-
-	if p, _ := strconv.Atoi(port); p < 1 || p > 65535 {
-		return fmt.Errorf("bad port number %s", port)
-	}
-	return nil
-}
-
-// validateNameServer validates a configured name server and
-// appends a ":53" if a :port is not included. IPv6 addresses are also
-// enclosed in square brackets if not already so enclosed.
-func validateNameServer(nameServer string) (string, error) {
-	if ip := net.ParseIP(nameServer); ip != nil {
-		if ip.To4() != nil {
-			// IPv4 without port
-			nameServer = nameServer + ":53"
-		} else {
-			// IPv6 without port
-			nameServer = "[" + nameServer + "]:53"
-		}
-		return nameServer, nil
-	}
-	// Assume it's IP:port
-	err := validateHostAndPort(nameServer)
-	return nameServer, err
-}
-
 func (d *KubeDNSServer) startSkyDNSServer() {
 	glog.V(0).Infof("Starting SkyDNS server (%v:%v)", d.dnsBindAddress, d.dnsPort)
 	skydnsConfig := &server.Config{
 		Domain:  d.domain,
 		DnsAddr: fmt.Sprintf("%s:%d", d.dnsBindAddress, d.dnsPort),
 	}
-	if d.nameServers != "" {
```

Comment:
It will get initialized when kubeDNS initialize its configuration through Once.
Let's say the user provides nameservers as commandline parameters. In the case of using configMap or configDir, the nameserver from command line will be overwriten when kubedns initializes its config through sync once. As for no configmap or configdir, both kubedns and skydns will use the nameserver from commandline parameter after kubedns sync once. 
https://github.com/grayluck/dns/blob/0aaf6028e388502bf6c7948d0cdd02047c55daaa/cmd/kube-dns/app/server.go#L73